File: TrainingMathMAUI/TimerClass.cs
using System;

public class TimeError : Exception
{
    public TimeError(string? message) : base(message)
    {
    }
}

public class Timer
{
    private double? _startTime;

    public Timer()
    {
        _startTime = null;
    }

    public void Start()
    {
        // Starts a new timer
        if (_startTime.HasValue)
        {
            throw new TimeError("The timer is already counting...");
        }

        _startTime = GetCurrentTime();
    }

    public string Stop()
    {
        // Stops the timer
        if (!_startTime.HasValue)
        {
            throw new TimeError("Timer didn't start...");
        }

        double elapsedTime = GetCurrentTime() - _startTime.Value;
        _startTime = null;

        return $"{elapsedTime:F4}";
    }

    private double GetCurrentTime()
    {
        return (double)DateTime.Now.Ticks / TimeSpan.TicksPerSecond;
    }
}

File: math_it_flutter/python/timer_class.py
import time

class TimeError(Exception):
    """A custom exception used to report errors in use of Timer class"""

class Timer():
    def __init__(self) -> None:
        self._start_time = None

    def start(self):
        """Starts a new timer"""
        if self._start_time is not None:
            raise TimeError('The timer is already counting...')
        
        self._start_time = time.perf_counter()

    def stop(self):
        """Stops the timer"""
        if self._start_time is None:
            raise TimeError("Timer didn't started...")
        
        elapsed_time = time.perf_counter() - self._start_time
        self._start_time = None
        return f"{elapsed_time:.4f}"
    